Description: Point javacc configuration at the jar files
 provided as part of the distribution.
 .
 Also fix javacc clean and generation which is broken upstream.
Author: James Page <james.page@ubuntu.com>
Forwarded: no

Index: lucene-solr/lucene/common-build.xml
===================================================================
--- lucene-solr.orig/lucene/common-build.xml	2012-05-29 17:55:56.482256690 +0100
+++ lucene-solr/lucene/common-build.xml	2012-05-29 17:56:18.574318554 +0100
@@ -138,7 +138,7 @@
   <property name="m2.repository.url" value="file://${maven.dist.dir}"/>
   <property name="m2.repository.private.key" value="${user.home}/.ssh/id_dsa"/>
 
-  <property name="javacc.home" location="${common.dir}"/>
+  <property name="javacc.home" location="/usr/share/java"/>
   <property name="jflex.home" location="${common.dir}"/>
 
   <path id="jflex.classpath">
@@ -152,7 +152,7 @@
 
   <path id="javacc.classpath">
     <fileset dir="${javacc.home}/">
-      <include name="bin/lib/*.jar"/>
+      <include name="javacc.jar"/>
     </fileset>
   </path>
 
Index: lucene-solr/lucene/build.xml
===================================================================
--- lucene-solr.orig/lucene/build.xml	2012-05-29 17:55:56.502256746 +0100
+++ lucene-solr/lucene/build.xml	2012-05-29 17:57:46.914565976 +0100
@@ -514,7 +514,7 @@
   <!-- ================================================================== -->
   <target name="clean-javacc">
     <delete>
-      <fileset dir="src/java/org/apache/lucene/queryParser" includes="*.java">
+      <fileset dir="core/src/java/org/apache/lucene/queryParser" includes="*.java">
         <containsregexp expression="Generated.*By.*JavaCC"/>
       </fileset>
       <fileset dir="contrib/queryparser/src/java/org/apache/lucene/queryParser/surround/parser" includes="*.java">
@@ -526,19 +526,19 @@
     </delete>
   </target>
 
-  <target name="javacc" depends="javacc-check,clean-javacc,javacc-QueryParser,javacc-contrib-queryparser,javacc-contrib-demo"/>
+  <target name="javacc" depends="javacc-check,clean-javacc,javacc-QueryParser,javacc-contrib-queryparser,javacc-contrib-benchmark"/>
 
   <target name="javacc-QueryParser" depends="javacc-check" if="javacc.present">
     <sequential>
-      <invoke-javacc target="src/java/org/apache/lucene/queryParser/QueryParser.jj"
-                     outputDir="src/java/org/apache/lucene/queryParser"/>
+      <invoke-javacc target="core/src/java/org/apache/lucene/queryParser/QueryParser.jj"
+                     outputDir="core/src/java/org/apache/lucene/queryParser"/>
 
       <!-- Change the inccorrect public ctors for QueryParser to be protected instead -->
-      <replaceregexp file="src/java/org/apache/lucene/queryParser/QueryParser.java"
+      <replaceregexp file="core/src/java/org/apache/lucene/queryParser/QueryParser.java"
 		     byline="true"
 		     match="public QueryParser\(CharStream "
 		     replace="protected QueryParser(CharStream "/>
-      <replaceregexp file="src/java/org/apache/lucene/queryParser/QueryParser.java"
+      <replaceregexp file="core/src/java/org/apache/lucene/queryParser/QueryParser.java"
 		     byline="true"
 		     match="public QueryParser\(QueryParserTokenManager "
 		     replace="protected QueryParser(QueryParserTokenManager "/>
@@ -553,6 +553,13 @@
     />
   </target>
 
+  <target name="javacc-contrib-benchmark" depends="javacc-check" if="javacc.present">
+    <ant target="javacc"
+      dir="contrib/benchmark"
+      antfile="build.xml"
+    />
+  </target>
+
   <target name="javacc-contrib-demo" depends="javacc-check" if="javacc.present">
     <ant target="javacc"
       dir="contrib/demo"
